components/org.wso2.carbon.identity.data.publisher.authentication.audit/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/audit/AuthenticationAuditLoggerUtils.java

```diff
@@ -0,0 +1,257 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.data.publisher.authentication.audit;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.application.authentication.framework.AuthenticatorStatus;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.StepConfig;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.context.SessionContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedIdPData;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.data.publisher.authentication.audit.model.AuthenticationAuditData;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.event.Event;
+
+import java.util.Map;
+
+/**
+ * Utilities for authentication audit logger.
+ */
+public class AuthenticationAuditLoggerUtils {
+
+    /**
+     * Create authentication data object from event for respective authentication step.
+     *
+     * @param event - triggered event
+     * @param authType - authentication type
+     * @return populated AuthenticationAuditData object
+     */
+    public static AuthenticationAuditData createAuthenticationAudiDataObject(Event event, String authType) {
+
+        Map<String, Object> properties = event.getEventProperties();
+        AuthenticationContext context = getAuthenticationContextFromProperties(properties);
+        Map<String, Object> params = getParamsFromProperties(properties);
+        AuthenticatorStatus status = getAuthenticatorStatusFromProperties(properties);
+
+        AuthenticationAuditData authenticationAuditData = new AuthenticationAuditData();
+
+        authenticationAuditData.setContextIdentifier(getContextIdentifier(context));
+        authenticationAuditData.setServiceProvider(getServiceProvider(context));
+        authenticationAuditData.setInboundProtocol(getInboundProtocol(context));
+        authenticationAuditData.setRelyingParty(getRelyingParty(context));
+
+        if (authType.equals(AuthenticationAuditLoggerConstants.AUDIT_AUTHENTICATION_STEP)) {
+            authenticationAuditData.setAuthenticatedUser(getUserNameForAuthenticationStep(params));
+            authenticationAuditData.setTenantDomain(getTenantDomainForAuthenticationStep(params));
+            authenticationAuditData.setAuthenticatedIdps(getIdentityProviderForAuthenticationStep(context));
+            authenticationAuditData.setStepNo(getStepNoForAuthenticationStep(context));
+
+        } else if (authType.equals(AuthenticationAuditLoggerConstants.AUDIT_AUTHENTICATION)) {
```

Comment:
What if authType is null ? Usually this is evaluated other way around. Or else you can use stringUtils